data_pipeline/google_analytics/ga_pipeline.py

```diff
@@ -59,6 +59,34 @@ def transform_response_to_bq_compatible_record(
             yield bq_json_formatted_record
 
 
+# pylint: disable=too-many-arguments
+def iter_get_report_pages(
+        analytics, metrics: list, dimensions: list,
+        ga_config, from_date: str, to_date: str = None
+):
+    page_token = None
+    while True:
+        response = analytics.get_report(
+            date_ranges=[{'startDate': from_date, 'endDate': to_date}],
+            view_id=ga_config.ga_view_id,
+            metrics=metrics,
+            dimensions=dimensions,
+            page_token=page_token
+        )
+
+        reports = response.get('reports', [])
+        # api can return multiple reports, however,
+        # this implementation considers single report
+        page_token = (
+            reports[0].get('nextPageToken')
+            if len(reports) == 1 else None
+        )
+        if not page_token:
+            yield response
+            break
+        yield response
```

Comment:
minor: I guess the yield could be moved above the if statement (then there would be just one)

Is it possible to have no pages? I guess it would just be a page with no items?

Reply:
if there is no next page, `page_token` changes to 'None', and when `page_token` is `None`, then you start all over again from the first page

Reply:
I mean, what if there is no content / page at all. Currently we will always yield at least one page.

Reply:
you will get a response page with no data content